browser/net/decentralized_dns_network_delegate_helper_unittest.cc

```diff
@@ -112,58 +133,104 @@ TEST_F(DecentralizedDnsNetworkDelegateHelperTest,
   EXPECT_TRUE(brave_request_info->new_url_spec.empty());
 }
 
+TEST_F(DecentralizedDnsNetworkDelegateHelperTest,
+       DecentralizedDnsPreRedirectTLDs) {
+  local_state()->SetInteger(kUnstoppableDomainsResolveMethod,
+                            static_cast<int>(ResolveMethodTypes::ETHEREUM));
+  const char* valid_urls[] = {
+      "https://brave.crypto", "https://brave.x",
```

Comment:
Would we expect that a different protocol scheme would ever be used with unstoppable domains. Can we add some tests for those cases if so

Reply:
Purpose of this test is to cover these checks:
https://github.com/brave/brave-core/blob/04732b0cb16448c3d46fbabcd1ba1daec87d55d1/browser/net/decentralized_dns_network_delegate_helper.cc#L37
https://github.com/brave/brave-core/blob/e6227e93eb7723b9ea51a7a19be0cc212cda4a69/components/decentralized_dns/utils.cc#L38
So this is not about protocols support, but about proper tlds.